### PR TITLE
UI changes (2/2)

### PR DESCRIPTION
**Pull request by rueter37**
This PR does the following changes:
- Fix positioning of the item count/skill cost in the Targetstatus window.
- The stats in the skill selection scene are updated now if the user leaves the ActorTarget scene.
- Change spacing and add arrows in the Savegame window.
- Moved HP a bit to the right in the Battlestatus and Skillstatus window.
- Some labels have been realigned in the RPG Maker 2003 ActorInfo window.
- Faces have been moved left in the RPG Maker 2003 main menu window.
- Fix cropped selection boxes in message windows.

Fixes: #2216
Fixes: #1617

### CODE REVIEW

**Review by Ghabry — CHANGES_REQUESTED** [review]
as discussed in the other PR: Please readd https://github.com/EasyRPG/Player/pull/2313/commits/d9a23bba377b341d9116d036ab2e164aa5884516 but hide it behind a boolean flag that is false by default

**Comment by Ghabry** [comment]
oh and I just remember another long-standing UI bug with the selectable window.

You can see this in "Unterwegs in Düsterburg": Start the game (skip intro), talk to your dog and when the selection window appears the selection box is cropped because there is Grandys face at the right.

**Comment by rueter37** [comment]
Endless scrolling can be disabled now.

**Comment by rueter37** [comment]
The selection box should be fixed now.

**Comment by rueter37** [comment]
This PR should no longer interfere with #2399. It can be reviewed anytime now.

**Comment by rueter37** [comment]
The requested changes should be addressed a long time ago. But maybe I missed something.

**Review by fdelapena — APPROVED** [review]

**Comment by fdelapena** [comment]
#2399 should be merged first because window_battlestatus.cpp has potential conflicts on nearer lines and likely would cause rebase hell.

**Comment by rueter37** [comment]
Okay, I'm going to rebase this one as soon as #2399 is merged in. :+1:

**Comment by rueter37** [comment]
Did the rebase now.

**Review by fdelapena — APPROVED** [review]

**Comment by fdelapena** [comment]
jenkins: test this please

File: src/window_savefile.cpp
Comment (Ghabry) [inline src/window_savefile.cpp:116]: btw this stringstream stuff is old legacy code. If you want you can replace it with appropriate fmt::format calls

File: src/scene_skill.cpp
Comment (Ghabry) [inline src/scene_skill.cpp:49]: prev_scene is unused here, simply delete the param name

File: src/window_selectable.cpp
Comment (Ghabry) [inline src/window_selectable.cpp:138]: You check twice for IsTriggered which was already done 2 lines earlier. (same for DOWN)
Why is that necessary?

**Comment by Ghabry** [comment]
Commits pending review by me:

- Targetstatus: Fix skill cost and item count display
- Savefile window: Fix distance between file and index number 
- Battlestatus and Skillstatus window: Moved HP a bit to the right

The others were reviewed.

File: src/scene_skill.cpp
Comment (rueter37) [reply src/scene_skill.cpp:49]: Well, the `Continue` function expects that the `prev_scene` parameter is present even if it is unused, otherwise compiling fails. Or did I miss something?

Comment (fmatthew5876) [reply src/scene_skill.cpp:49]: He means leave the parameter, but delete it's name. Otherwise you get a compiler warning about unused variables

File: src/window_selectable.cpp
Comment (rueter37) [reply src/window_selectable.cpp:138]: Because if endless scrolling is disabled, the second check ensures that you still can jump to the begin or end of the menu if you press up or down while not holding the key down.

File: src/scene_skill.cpp
Comment (rueter37) [reply src/scene_skill.cpp:49]: So this means to change this to `void Scene_Skill::Continue(SceneType)` in `scene_skill.cpp` while leaving this in `scene_skill.h` unchanged? (I'm probably asking too much dumb questions)

File: src/window_savefile.cpp
Comment (rueter37) [reply src/window_savefile.cpp:116]: There are even more spots which use the stringstreams but I better not touch them because of the risk to break the `override_name` stuff in the savegames.

File: src/scene_skill.cpp
Comment (Ghabry) [reply src/scene_skill.cpp:49]: yeah keep in h, delete in cpp

Comment (rueter37) [reply src/scene_skill.cpp]: Fixed.

File: src/window_actorstatus.cpp
Comment (carstene1ns) [inline src/window_actorstatus.cpp:49]: Can we fix up the spacing between `DrawMinMax` function arguments?
I know it has been in there before, however it looks ugly.

File: src/window_actorstatus.cpp
Comment (carstene1ns) [inline src/window_actorstatus.cpp:48]: Double ternary is not easy to read IMO. This could be changed to a function lambda I guess, since almost the same calculation is done for HP and SP.

**Review by carstene1ns — APPROVED** [review]

File: src/window_actorstatus.cpp
Comment (rueter37) [reply src/window_actorstatus.cpp]: Fixed.

File: src/window_actorstatus.cpp
Comment (rueter37) [reply src/window_actorstatus.cpp:48]: Please excuse my dumb question, but what does "function lambda" mean?

Comment (carstene1ns) [reply src/window_actorstatus.cpp:48]: untested proposal:

```c++
auto get_fontcolor() = [](int have, int max, bool can_knockout) {

	if (can_knockout && have == 0) return Font::ColorKnockout;

	if (max > 0 && (have <= max / 4)) return Font::ColorCritical;

	return Font::ColorDefault;

}
```

Use like this:

```c++
get_fontcolor(actor->GetHp(), actor->GetMaxHp(), true)
```
```c++
get_fontcolor(actor->GetSp(), actor->GetMaxSp(), false)
```

What I dislike here is that the arguments are duplicated in the `DrawMinMax()` call then.

Comment (carstene1ns) [reply src/window_actorstatus.cpp:48]: maybe even with captures:

```diff
diff --git a/src/window_actorstatus.cpp b/src/window_actorstatus.cpp
index b716b994..95b4c640 100644
--- a/src/window_actorstatus.cpp
+++ b/src/window_actorstatus.cpp
@@ -43,15 +43,24 @@ void Window_ActorStatus::DrawStatus() {
 
 	Game_Actor* actor = Main_Data::game_actors->GetActor(actor_id);
 
+	int have, max;
+	auto fontcolor = [&have, &max](bool can_knockout) {
+		if (can_knockout && have == 0) return Font::ColorKnockout;
+		if (max > 0 && (have <= max / 4)) return Font::ColorCritical;
+		return Font::ColorDefault;
+	};
+
 	// Draw Hp
 	contents->TextDraw(1, 2, 1, lcf::Data::terms.health_points);
-	int color = (actor->GetHp() == 0 ? Font::ColorKnockout : (actor->GetHp() <= actor->GetMaxHp() / 4 ? Font::ColorCritical : Font::ColorDefault));
-	DrawMinMax(90, 2, actor->GetHp(), actor->GetMaxHp(), color);
+	have = actor->GetHp();
+	max = actor->GetMaxHp();
+	DrawMinMax(90, 2, have, max, fontcolor(true));
 
 	// Draw Sp
 	contents->TextDraw(1, 18, 1, lcf::Data::terms.spirit_points);
-	color = (actor->GetMaxSp() > 0 && actor->GetSp() <= actor->GetMaxSp() / 4 ? Font::ColorCritical : Font::ColorDefault);
-	DrawMinMax(90, 18, actor->GetSp(), actor->GetMaxSp(), color);
+	have = actor->GetSp();
+	max = actor->GetMaxSp();
+	DrawMinMax(90, 18, have, max, fontcolor(false));
 
 	// Draw Exp
 	contents->TextDraw(1, 34, 1, lcf::Data::terms.exp_short);
```

Comment (rueter37) [reply src/window_actorstatus.cpp]: Your proposal works for me. I have added an extra commit which contains these changes.

File: src/window_selectable.cpp
Comment (Ghabry) [reply src/window_selectable.cpp:138]: Imo this if-stuff is quite ugly. Alternative patch (apply this to your HEAD and then squash this commit and my patch together): https://gist.github.com/Ghabry/cc1a20c3b31f461a5cf2a51e1519b9ab

Changes:
``disable_endless_scrolling`` is now ``endless_scrolling`` because double negation is bad (``!disable_endless_scrolling -> endless_scrolling``).
The move up and down logic is extracted in a lambda and the branches switched between "triggered" and "repeating".

Please test if I missed any case for "endless_scrolling = false".

Comment (rueter37) [reply src/window_selectable.cpp]: Thanks for the patch, it works for me. It is included now.

**Review by Ghabry — APPROVED** [review]

**Pull request merged**
